rest/route/subscription_routes.go

```diff
@@ -0,0 +1,84 @@
+package route
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/rest"
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/evergreen/util"
+)
+
+type subscriptionPostHandler struct {
+	Subscriptions   *[]model.APISubscription `json:"subscriptions"`
+	dbSubscriptions []event.Subscription
+}
+
+func getSubscriptionRouteManager(route string, version int) *RouteManager {
+	h := &subscriptionPostHandler{}
+
+	handler := MethodHandler{
+		PrefetchFunctions: []PrefetchFunc{PrefetchUser},
+		Authenticator:     &RequireUserAuthenticator{},
```

Comment:
Do we need to auth a specific user (to match subscriptions with that user), or is that out of scope for this change?

Reply:
Added a check and added a test

Reply:
lgtm